Improve backtesting documentation

xmatthias opened the pull request:
## Summary
Explain in one sentence the goal of this PR

Explain problems raised in #2277 

## Quick changelog

- Seperate download-data documentation into it's own file
- explain assumptions backtesting makes
- improve wording in backtesting documentation
- link to download-data from hyperopt

hroff-1902 commented on docs/hyperopt.md:
```diff
@@ -6,6 +6,9 @@ algorithms included in the `scikit-optimize` package to accomplish this. The
 and still take a long time.
 
+Hyperopt requires historic data to be available, just as backtesting does.
+To learn how to get data for the pairs and exchange you're interrested in, head over to the [Data downloading](data-download.md) section of the documentation.
```
The section named "Data download" in its file. Maybe "Data downloading" is even better. But should be same in all places.

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -107,31 +74,33 @@ freqtrade backtesting --export trades --export-filename=backtest_samplestrategy.
-Doing `--timerange=-200` will get the last 200 timeframes
-from your inputdata. You can also specify specific dates,
-or a range span indexed by start and stop.
+Doing `--timerange=20190101-` will all available data starting with January 1st, 2019 from your inputdata.
```
will do what? (no predicate in the sentence)

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -107,31 +74,33 @@ freqtrade backtesting --export trades --export-filename=backtest_samplestrategy.
-from your inputdata. You can also specify specific dates,
-or a range span indexed by start and stop.
+Doing `--timerange=20190101-` will all available data starting with January 1st, 2019 from your inputdata.
+You can also specify specific dates, or a range span indexed by start and stop.
```
* "specify particular dates", "specify concrete dates"

* no comma before "or"

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -46,14 +14,13 @@ real data. This is what we call
 
-The result of backtesting will confirm you if your bot has better odds of making a profit than a loss.
-
-The backtesting is very easy with freqtrade.
```
excellent! (that you've removed this sentence) 👏 

Backtesting is not easy by its nature...

xmatthias replied:
backtesting itself is easy - it's having a good strategy that's hard :D 

no matter, this has no place in a documentation i think ... not like this, anyway

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -1,41 +1,9 @@
-- Use `--timeframes` to specify which tickers to download. Default is `--timeframes 1m 5m` which will download 1-minute and 5-minute tickers.
-- To use exchange, timeframe and list of pairs as defined in your configuration file, use the `-c/--config` option. With this, the script uses the whitelist defined in the config as the list of currency pairs to download data for and does not require the pairs.json file. You can combine `-c/--config` with most other options.
+Backtesting requires historic data to be available.
+To learn how to get data for the pairs and exchange you're interrested in, head over to the [Data downloading](data-download.md) section of the documentation.
```
* "interested"

* The section named "Data download" in its file. Maybe "Data downloading"/"Downloading data" is even better. But should be same in all places.

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -46,14 +14,13 @@ real data. This is what we call
-ask you to download them first using `freqtrade download-data`. 
-For details on downloading, please refer to the [relevant section](#Getting-data-for-backtesting-and-hyperopt) in the documentation.
+ask you to download them first using `freqtrade download-data`.
+For details on downloading, please refer to the [Downloading data](data-download.md) section in the documentation.
```
The section named "Data download" in its file. Maybe "Data downloading"/"Downloading data" is even better. But should be same in all places.

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -215,22 +179,33 @@ time a trade will reach 1%).
+- Trailing stoploss
+  - High happens first - adjusting stoploss
+  - Low uses the adjusted stoploss (so sells with large high-low difference are backtested correctly)
+- Sell-reason does not explain if a trade was positive or negative, just what triggered the sell (this can look odd if negative ROI values are used)
```
Should we allow it (negative roi)? -- not for this PR though...
Moreover, should we describe this here?

xmatthias replied:
it is technically allowed now - and it allows to set a "timer" on trades, forcing a sell after X minutes.

thats not how it's intendet to be used - but limiting ROI to positive values would increase the config effort (and code needed) to allow such a case:
you'd need the same number of minutes in ROI (for positive trade) and stoploss (for negative trades) ... and honestly, i don't want to add even more functionality into stoploss - it's already way too complicated with all the different options.

hroff-1902 replied:
I mean, should we even mention in the docs that this can be negative? To avoid questions it may cause...

xmatthias replied:
since it's allowed, and obviously people try to use it, i would do this, yes.

it maybe should be documented clearer (not in this PR ...) - as to why this configuration may make sense (limit max duration of a trade)...

xmatthias commented:
Reviewing documentation is almost never easy (code is easier to review, to me at least...) - especially if parts were split ... 

I think i've addressed all feedbacks ... it's a lot of comments to go through for 4 changed files :laughing:

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -84,8 +84,8 @@ freqtrade backtesting --timerange=20190501-
 
```
Is not it the same with that written below in the lines 85-88?
Why the example is here but new subsection is started at the line 85?

xmatthias replied:
This is not clear to me ... ?

hroff-1902 replied:
I mean here that now, after the change, the line 82 contains
`freqtrade backtesting --timerange=20190501-`

It's completely same usage of the option with the examples below, in lines 87-104. But this one is in another subsection since a new subsection starts at the line 85.

The point is do we really need this in two different subsections, here above (line 82) AND below (lines 87-104)?

xmatthias replied:
I'm not certain - the first is a quick command "with smaller testset" ... while the next section is advanced usage ... 

but yes, we could also combine the 2 i guess

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -84,8 +84,8 @@ freqtrade backtesting --timerange=20190501-
 
-Doing `--timerange=20190101-` will all available data starting with January 1st, 2019 from your inputdata.
-You can also specify specific dates, or a range span indexed by start and stop.
+Using `--timerange=20190101-` will use all available data starting with January 1st, 2019 from your inputdata.
```
using...will use

"Example: Running backtesting with the `--timerange=20190101-` option will use all available..."?

^ "Example: " (or "For example, ") is missing?

hroff-1902 commented on docs/data-download.md:
```diff
@@ -0,0 +1,58 @@
+# Data download
```
"Data Downloading", as in the other places referring to this section.

hroff-1902 commented on docs/backtesting.md:
```diff
@@ -84,7 +84,7 @@ freqtrade backtesting --timerange=20190501-
 #### Advanced use of timerange
 
-Using `--timerange=20190101-` will use all available data starting with January 1st, 2019 from your inputdata.
+For Example, running backtesting with the `--timerange=20190101-` option will use all available data starting with January 1st, 2019 from your inputdata.
```
"For example, "

hroff-1902 commented:
Fine, finally that's what I meant...
